src/parser/transform/statement/transform_create_table.cpp

```diff
@@ -107,7 +108,14 @@ unique_ptr<CreateStatement> Transformer::TransformCreateTable(duckdb_libpgquery:
 			break;
 		}
 		case duckdb_libpgquery::T_PGConstraint: {
-			info->constraints.push_back(TransformConstraint(c));
+			auto cond = TransformConstraint(c);
+			if (cond->type == ConstraintType::FOREIGN_KEY) {
+				auto &fk = (ForeignKeyConstraint &)*cond;
+				if (fk.pk_table == info->table) {
+					throw ParserException("Main key Table must be different with foreign key table");
```

Comment:
Great, it seems this is good feature.
But I think it would be a little tricky to implement this feature in the current implementation.
Can't we implement the current first and add it afterwards?

Reply:
Okay, I will add this feature.